### ISSUE TITLE
Community Detection Demo GraphSAGELinkGenerator Issue

### ISSUE THREAD

**Issue by dubbsbrandon** [stellargraph/stellargraph#737]
### Describe the bug

Training generator produced from GraphSAGELinkGenerator is of type stellargraph.mapper.sequences.OnDemandLinkSequence when it should be GraphSAGENodeGenerator or GraphSAGELinkGenerator

### To Reproduce

To reproduce, follow the demo to cell 37, which will result in the error as the training generator is being created.

### Observed behavior

![Screen Shot 2020-01-28 at 4 04 43 PM](https://user-images.githubusercontent.com/54581642/73305394-ff9dbd80-41e7-11ea-9305-94c1673c59ea.png)

### Expected behavior

Link generator to be created.

**Comment by huonw** [comment]
Hi @dubbsbrandon, thanks for using the library!

Could you say a bit more about the environment you're working in? In particular:

- what version of stellargraph are you using (`print(stellargraph.__version__)` in Python or `pip freeze` in a terminal)?
- how did you get the demo? E.g. did you download it from GitHub?

**Comment by dubbsbrandon** [comment]
I just pulled the code directly from the examples, line by line. Nothing implemented myself. The version I have is the 0.9.0b version. Is there another, more updated one?

**Comment by huonw** [comment]
> The version I have is the 0.9.0b version.

Ah, from that I'm guessing that you cloned the repository with `git` and ran something like `pip install .`? And then ran the examples out of the cloned directory too. Thanks for the info, it helps eliminate a possibility of other problems, like a version mismatch between the `stellargraph` library and which examples you're running 👍 

I can reproduce the problem, and I've put a fix for it at https://github.com/stellargraph/stellargraph/pull/747. I'm still verifying that it completely fixes the demo, but it at least fixes the problem here.

**Comment by huonw** [comment]
That PR has been merged as 4d02ca3, so updating (e.g. `git pull` along with: `pip install --upgrade .` if it wasn't an `-e`/`--editable` install) should get the example working again.

Alternatively you can just apply the two cell fix there. Change:

```python
train_gen = GraphSAGELinkGenerator(Gs, batch_size, num_samples).flow(unsupervisedSamples)
```

to

```python
generator = GraphSAGELinkGenerator(Gs, batch_size, num_samples)
train_gen = generator.flow(unsupervisedSamples)
```
and

```python
layer_sizes=layer_sizes, generator=train_gen, bias=True, dropout=0.0, normalize="l2"
```

to

```python
layer_sizes=layer_sizes, generator=generator, bias=True, dropout=0.0, normalize="l2"
```

**Comment by dubbsbrandon** [comment]
I meant to followup last week, but everything works fine all the way through. You can probably consider this closed unless there is something else I'm unaware of.
